src/GitTools.IssueTrackers/AuthSettings.cs

```diff
@@ -0,0 +1,38 @@
+namespace GitTools.IssueTrackers
+{
+    using System;
+    using JetBrains.Annotations;
+
+    public class AuthSettings
+    {
+        public AuthSettings()
+        {
+            IsEmpty = true;
+        }
+
+        public AuthSettings([NotNull] string username, [NotNull] string password)
+        {
+            if (string.IsNullOrEmpty(username)) throw new ArgumentNullException("username");
+            if (string.IsNullOrEmpty(password)) throw new ArgumentNullException("password");
+            Username = username;
+            Password = password;
+            IsUsernameAndPasswordAuthentication = true;
+        }
+
+        public AuthSettings([NotNull] string token)
+        {
+            if (string.IsNullOrEmpty(token)) throw new ArgumentNullException("token");
+            Token = token;
+            IsTokenAuthentication = true;
+        }
+
+        public bool IsEmpty { get; private set; }
+
+        public string Token { get; private set; }
+        public bool IsTokenAuthentication { get; private set; }
```

Comment:
This class is immutable at the moment, we have no need to recalculate and these can be set straight off the ctor.